scripts/release/README.md

```diff
@@ -65,10 +71,10 @@ If the OTP code expires while publishing, re-run this command and answer "y" to
 
 Stable releases should always be created from the "next" channel. This encourages better testing of the actual release artifacts and reduces the chance of unintended changes accidentally being included in a stable release.
 
-To prepare a stable release, choose a "next" version and run the [`prepare-stable`](#prepare-stable) script <sup>1</sup>:
+To prepare a stable release, choose a "next" version and run the [`promote-to-stable`](#promote-to-stable) script <sup>1</sup>:
 
 ```sh
-scripts/release/prepare-stable.js --version=0.0.0-241c4467e
+scripts/release/promote-to-stable.js --version=0.0.0-241c4467e
```

Comment:
Can we call this `download-stable` or something? Since it corresponds to `download-release`, only difference being that it downloads the tarballs from npm.

I was thinking at some point we should consolidate these scripts and use the `download-release` process for stable releases, too — i.e. always download from CI artifacts. The rationale for downloading from npm was to reduce the amount of processing in between the prerelease tarball and the final release tarball. But since they all come from the CI tarball, anyway, it actually seems safer to me to use the same script for both types of releases. Then the only additional point of failure is what versions are given to the version rewriting step.

Reply:
Happy to change the name but I don't like that one because the script doesn't download a *stable* release. It downloads a *next* release, and prepares it to be published as stable.

> I was thinking at some point we should consolidate these scripts and use the download-release process for stable releases, too — i.e. always download from CI artifacts. The rationale for downloading from npm was to reduce the amount of processing in between the prerelease tarball and the final release tarball. But since they all come from the CI tarball, anyway, it actually seems safer to me to use the same script for both types of releases. Then the only additional point of failure is what versions are given to the version rewriting step.

The one thing the current approach gives us that you the one you mention wouldn't is that it forces us to release a *next* release first (and presumably test it) before then promoting it to stable.

Reply:
> the script doesn't download a stable release. It downloads a next release, and prepares it to be published as stable.

That's true implementation-wise, but from the user's perspective I don't think it matters because the end result is the same: a stable release in the `build` directory ready to be published

> The one thing the current approach gives us that you the one you mention wouldn't is that it forces us to release a next release first (and presumably test it) before then promoting it to stable.

Right, though you could still enforce that by pinging npm to confirm that a prerelease was  published.

Reply:
Maybe `prepare-next` and `prepare-stable`? I don't really care what the verb is but I do think it would be nice to be symmetrical

Reply:
> Maybe `prepare-`next and `prepare-stable`?

This is what we had before :wink: but it's not very descriptive/accurate, given that `prepare-next` is the script you used to both prepare a *next* release and an *experimental* release. I don't think we should have the NPM tag in the name unless the script is *only* used for that tag (like "stable")

Reply:
Maybe `prepare-stable` is still okay since that's what it's doing.

Reply:
Oh right :D

How about `prepare-release` versus `prepare-prerelease`? A bit awkward but avoids the "stable" versus "experimental" confusion

Reply:
Or we could go the other way and add a separate command for experimental:

- `prepare-stable`
- `prepare-next`
- `prepare-experimental`

Reply:
Or (sorry just thinking out loud) a single `prepare-release` command with a `channel` argument.

Reply:
Uh...sure, that sounds fine...I don't really want to re-read and update the README again though 😭

Reply:
It's fine. I can update it one more time.

Reply:
Maybe?
* `build-release-locally`: builds local source
* `prepare-release-from-ci`: prepares CI artifacts to be published as whatever
* `prepare-release-from-npm`: prepares "next" to be promoted to "stable"

Reply:
I don't know. In a way, it becomes harder to evaluate names the more possibilities we toss out. I think any fo the revisions I've pushed so far are an improvement over what was there before. I'd like to prefer making another major change (like combining the scripts) right now b'c it doesn't feel like the most valuable use of time.